### PR TITLE
feat: add publish to cast

### PR DESCRIPTION
**Pull request by ayushm2003**
Adds `publish` to cast letting users send raw transactions to the network.

### CODE REVIEW

File: cast/src/lib.rs [hunk 1]
```diff
@@ -146,6 +146,31 @@ where
...
+    /// println!("{:?}", res);
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn publish(&self, raw_tx: String) -> Result<PendingTransaction<'_, M::Provider>> {
+        let edit_raw_tx: String = match raw_tx.starts_with("0x") {
+            true => raw_tx.split_at(2).1.to_string(),
+            false => raw_tx,
+        };
+        let tx = Bytes::from(hex::decode(edit_raw_tx)?);
```
Comment (onbjerg) [inline cast/src/lib.rs:168]: I think this is more idiomatic:

```suggestion
    pub async fn publish(&self, mut raw_tx: String) -> Result<PendingTransaction<'_, M::Provider>> {
        raw_tx = match raw_tx.strip_prefix("0x") {
            Some(s) => s.to_string(),
            None => raw_tx,
        };
        let tx = Bytes::from(hex::decode(raw_tx)?);
```

Comment (onbjerg) [reply cast/src/lib.rs:168]: Or even
```suggestion
    pub async fn publish(&self, raw_tx: String) -> Result<PendingTransaction<'_, M::Provider>> {
        let tx = match raw_tx.strip_prefix("0x") {
            Some(raw) => Bytes::from(hex::decode(raw)?),
            None => Bytes::from(hex::decode(raw_tx)?),
        };
```

**Comment by onbjerg** [comment]
Otherwise lgtm

File: cast/src/lib.rs (hunk 1 above)
Comment (ayushm2003) [reply cast/src/lib.rs:168]: thank you. I am still a noobie with rust so this is very helpful.

**Review by gakonst — APPROVED** [review]

**Pull request merged**